source/fundamentals/object-serialization.txt

```diff
@@ -0,0 +1,202 @@
+   MongoClient client = MongoClients.create(settings);
+
+All ``MongoDatabase`` objects created from the ``MongoClient``
+(and all ``MongoCollection`` objects created from those ``MongoDatabase``
+objects) use the customized ``CodecRegistry`` unless overridden.
+
+Use a Custom Codec with a Database
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``MongoDatabase`` objects inherit their ``CodecRegistry`` from the
+``MongoClient`` used to create them. You can create a new
+``MongoDatabase`` instance with a custom ``CodecRegistry`` from an
+existing ``MongoDatabase`` instance using the
+:java-sync-api:`MongoDatabase.withCodecRegistry()
+<com/mongodb/client/MongoDatabase.html#withCodecRegistry(org.bson.codecs.configuration.CodecRegistry)>`
+method.
+
+The following example creates a ``MongoDatabase`` with a
+``CodecRegistry`` containing the default codecs for ``Document``,
+``BasicDBObject``, and ``BsonDocument`` as well as a custom codec of
+type ``ExampleCodec``:
+
+.. code-block:: java
+
+   CodecRegistry codecRegistry =
+      CodecRegistries.fromRegistries(MongoClientSettings.getDefaultCodecRegistry(),
+                CodecRegistries.fromCodecs(new ExampleCodec()));
+   MongoDatabase database = client.getDatabase("exampleDatabase")
+                .withCodecRegistry(codecRegistry);
+
+All ``MongoCollection`` objects created from the ``MongoDatabase``
+use the customized ``CodecRegistry`` unless overridden.
+
+Use a Custom Codec with a Collection
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``MongoCollection`` objects inherit their ``CodecRegistry`` from the
+``MongoDatabase`` used to create them. You can create a new
+``MongoCollection`` instance with a custom ``CodecRegistry`` from an
+existing ``MongoCollection`` instance using the
+:java-sync-api:`MongoCollection.withCodecRegistry()
+<com/mongodb/client/MongoCollection.html#withCodecRegistry(org.bson.codecs.configuration.CodecRegistry)>`
+method.
+
+The following example creates a ``MongoCollection`` with a
+``CodecRegistry`` containing the default codecs for ``Document``,
+``BasicDBObject``, and ``BsonDocument`` as well as a custom codec of
+type ``ExampleCodec``:
+
+.. code-block:: java
+
+   CodecRegistry codecRegistry =
+      CodecRegistries.fromRegistries(MongoClientSettings.getDefaultCodecRegistry(),
+                fromCodecs(new ExampleCodec()));
+   MongoCollection<Example> collection = database.getCollection("exampleCollection")
+                .withCodecRegistry(codecRegistry);
+
+All queries executed on the ``MongoCollection`` use the customized
+``CodecRegistry``. The ``ExampleCodec`` serializes and deserializes
+objects of type ``Example``, allowing the ``MongoCollection`` to represent
+MongoDB documents with the ``Example`` class.
+
+Codec Providers
+---------------
+
+You can also enable serialization and deserialization for a class using
+a **codec provider**. A codec provider is a factory for codec instances.
+You can use a codec provider in the following ways:
```

Comment:
I think "use" doesn't fit the list items. Can you take another look at this section and its structure? The structure that I currently see is:
- codec provider definition (great, so far)
- use it the following ways:
-- non-default codec provider that is compatible with your class... (use it to do what?)
-- create your own codec provider... (not sure how this is using the codec provider as you're describing implementing the interface)
- add a provider using CodecRegistries.fromProviders() (add a provider to a CodecRegistry by creating a CodecRegistry?)

Reply:
Reworded to better match the new wording of the custom codec section.